CQ-week

- [x] Use cache for shop rules
- [x] Skip crawled pages of blacklisted shops.

Take a look at:

`Caused by: java.lang.IllegalArgumentException: json string can not be null or empty
May 25 19:16:37 BP2017N2WS03 Parser-1.0.0.jar[29971]:         at com.jayway.jsonpath.internal.Utils.notEmpty(Utils.java:386)
May 25 19:16:37 BP2017N2WS03 Parser-1.0.0.jar[29971]:         at com.jayway.jsonpath.internal.ParseContextImpl.parse(ParseContextImpl.java:36)
May 25 19:16:37 BP2017N2WS03 Parser-1.0.0.jar[29971]:         at com.jayway.jsonpath.JsonPath.parse(JsonPath.java:599)
May 25 19:16:37 BP2017N2WS03 Parser-1.0.0.jar[29971]:         at de.hpi.parser.service.DataExtractor.extract(DataExtractor.java:61)
`

This causes the parser to keep stuck within a return message and get it again loop.

--- a/src/main/java/de/hpi/parser/properties/AsyncConfig.java
+++ b/src/main/java/de/hpi/parser/properties/AsyncConfig.java
@@ -12,11 +12,11 @@ import java.util.concurrent.Executor;
 public class AsyncConfig {
 
     @Bean(name = "queueThreadPoolTaskExecutor")
-    public Executor queueThreadPoolTaskExecutor() {
+    public Executor queueThreadPoolTaskExecutor(ParserConfig config) {
         ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
-        executor.setCorePoolSize(4);
-        executor.setMaxPoolSize(100);
-        executor.setQueueCapacity(20);
+        executor.setCorePoolSize(config.getCorePoolSize());
+        executor.setMaxPoolSize(config.getMaxPoolSize());
+        executor.setQueueCapacity(config.getQueueCapacity());
         executor.setThreadNamePrefix("Queue-Threadpool-");
         executor.initialize();
         return executor;

--- a/src/main/java/de/hpi/parser/properties/ParserConfig.java
+++ b/src/main/java/de/hpi/parser/properties/ParserConfig.java
@@ -7,6 +7,8 @@ import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.retry.annotation.EnableRetry;
 import org.springframework.stereotype.Component;
 
+import javax.validation.constraints.Min;
+
 @Component
 @EnableConfigurationProperties
 @EnableRetry
@@ -20,5 +22,14 @@ public class ParserConfig {
     private String shopRulesGeneratorRoot;
 
     private String getRulesRoute;
+
+    @Min(1)
+    private int corePoolSize;
+
+    @Min(1)
+    private int maxPoolSize;
+
+    @Min(1)
+    private int queueCapacity;
 
 }

